src/Microsoft.AspNetCore.Rewrite/Internal/BackReferenceCollection.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal
+{
+    public class BackReferenceCollection
+    {
+        private List<string> backReferences = new List<string>();
```

Comment:
Yeah, exactly. Every time its created we immediately add something to the backReferences list so using lazy initialization at least to my knowledge wouldn't really help.